packages/core/src/errors/core_error.rs

```diff
@@ -127,65 +127,68 @@ impl StdError {
     }
 
     pub fn invalid_hex(msg: impl ToString) -> Self {
-        StdError::InvalidHex {
+        CoreError::InvalidHex {
             msg: msg.to_string(),
             backtrace: BT::capture(),
         }
     }
 
     pub fn invalid_utf8(msg: impl ToString) -> Self {
-        StdError::InvalidUtf8 {
+        CoreError::InvalidUtf8 {
             msg: msg.to_string(),
             backtrace: BT::capture(),
         }
     }
 
     pub fn not_found(kind: impl Into<String>) -> Self {
-        StdError::NotFound {
+        CoreError::NotFound {
             kind: kind.into(),
             backtrace: BT::capture(),
         }
     }
 
     pub fn parse_err(target: impl Into<String>, msg: impl ToString) -> Self {
-        StdError::ParseErr {
+        CoreError::ParseErr {
             target_type: target.into(),
             msg: msg.to_string(),
             backtrace: BT::capture(),
         }
     }
 
     pub fn serialize_err(source: impl Into<String>, msg: impl ToString) -> Self {
-        StdError::SerializeErr {
+        CoreError::SerializeErr {
             source_type: source.into(),
             msg: msg.to_string(),
             backtrace: BT::capture(),
         }
     }
 
     pub fn overflow(source: OverflowError) -> Self {
-        StdError::Overflow {
+        CoreError::Overflow {
             source,
             backtrace: BT::capture(),
         }
     }
 
     pub fn divide_by_zero(source: DivideByZeroError) -> Self {
-        StdError::DivideByZero {
+        CoreError::DivideByZero {
             source,
             backtrace: BT::capture(),
         }
     }
 }
 
-impl PartialEq<StdError> for StdError {
-    fn eq(&self, rhs: &StdError) -> bool {
+#[cfg(feature = "std")]
+impl std::error::Error for CoreError {}
```

Comment:
Intereesting, didn't know that `thiserror` registers the error types as the source automatically, I thought you'd need a separate attribute for that.

I'll look into that.

Reply:
Just switched back to pulling in `thiserror` when the `std` feature is activated.  
In general, the compile times might be *slightly* slower now because now `derive_more` and `thiserror` are both in the dependency tree. But it should be negligible